### ISSUE TITLE
Cache OSM Notes (for a few hours at least)

### ISSUE THREAD

**Issue by mnalis** [osmandapp/OsmAnd#12679]
# 🚀 feature request

### Description

When `OSM notes (online)` is enabled in `Configure map`, it makes requests all the time if data connection is available. This leads to huge battery drain; and if data connection is disabled, even all currently downloaded OSM notes disappear.

### Describe the solution you'd like

It would nice if OsmAnd would cache the downloaded OSM Notes for at least few hours (maybe even 24h or 48h?), like it does for online TMS/WMS tiles. One could then download notes for target region on WiFi and disable mobile data connection, to greatly extend battery usage and not accrue data charges. 

### Describe alternatives you've considered

Keeping the mobile data connection active at all times works, but:
- requires a device with mobile data option
- drains the battery like crazy (I guess every movement or screen resize sends new notes download request)
- incur data charges (esp. huge when roaming)

Related request: https://github.com/osmandapp/OsmAnd/issues/4424

**Comment by laralem** [comment]
Good idea. In many places most notes don't change / are not resolved, specially those notes needing checking in situ become very old.

**Comment by Lee-Carre** [comment]
Besides user-experience, reducing unnecessary load (via caching, or smarter code for deciding when to send API requests, or downloading & locally-storing the results until told to discard them (perma-cache?), or whatever) on OSM servers should be an important criterion, anyway. Being a good network citizen and all that. Especially with how popular OsmAnd is (many instances, all hammering the servers).

Having learned that OsmAnd is causing so much network activity, I'm strongly tempted to disable the feature in my own instance until the behaviour improves. Another concern (like the OP) is byte-count when using mobile data; in my area, tariffs are either not-cheap or not-generous.

In the meantime, might I suggest for other users to make a dedicated Surveying profile (which has the likes of OSM-note-fetching enabled) which is only used when actually surveying; other profiles being for general day-to-day navigation (with notes disabled). That might be a reasonable workaround / compromise, for the time being.

If a cache is implemented, then it would be helpful to have some way of causing OsmAnd to treat it as stale (at least for the area within the viewport) so that it re-checks with the server.
Ideally the duration of freshness should be user-configurable. Else, add a user-configuration variable to only refresh the cache when connected via 802.11; that would seem to do the right thing regardless of how long someone is using OsmAnd out and about (compare a daily commuter vs. a month-long adventure in the wilderness). I don't really see why anyone would want the cache to expire before they have suitable network access again; that would just be frustrating.

For reference / comparison, Vespucci has a quite usable & effective approach, as does StreetComplete. Specifically for QA, OsmBugs uses a simple yet effective design.
